app/src/main/java/com/github/geohunt/app/data/local/LocalPicture.kt

```diff
@@ -10,20 +9,8 @@ import java.io.File
  * This data class abstracts over regular files to ensure correctness at all times of the photo
  * files (e.g. extension, permission to read, etc.).
  *
- * @param file The underlying file that represents the image.
+ * @param uri The URI
  */
 data class LocalPicture(
-    val file: File
-) {
-    val uri: Uri = Uri.fromFile(file)
-
-    /**
-     * Returns true if the picture is valid, i.e. it can be safely read as an image.
-     * This function should be called before any read or upload.
-     */
-    fun isValid(): Boolean = file.isFile && file.canRead()
-
-    init {
-        require(isValid())
-    }
-}
\ No newline at end of file
+    val uri: Uri
+)
```

Comment:
I know I am being a massive hypocrite by saying that but if you could add a little constructor to support back compatibility :eyes:

Reply:
Done